app/scripts/lib/util/chrome-util.js

```diff
@@ -35,7 +42,17 @@ module.exports = {
 
   syncGetValue(key) {
     return promised((cb) => {
-      chrome.storage.sync.get(key, (items) => cb(items.hasOwnProperty(key) ? items[key] : undefined));
+      chrome.storage.sync.get(key, (items) => {
+        try {
+          cb(items.hasOwnProperty(key) ? items[key] : undefined);
+        } catch (err) {
+          // Firefox causes an security error when storage.local is accessed from popup scripts.
+          // https://bugzilla.mozilla.org/show_bug.cgi?id=1258139
+          // Perhaps this hack can be removed when storage.sync is officially supported, or the above bug is resolved.
```

Comment:
Thank you for trying! That seems a tough problem to fix... :disappointed:

> Is it better to wait until bug 1258139 is fixed?

No, I'll merge this anyway.
Custom rule setting is one of key features of this extension though, so I think we should keep calling it "experimental" until it gets fixed.